test/test_hyperparameters.py

```diff
@@ -538,6 +708,153 @@ def test_normalint_is_legal(self):
+                         [12, 9, 8, 11])
+
+        b1_log = BetaIntegerHyperparameter(
+            "param", lower=1.0, upper=1000.0, alpha=3.0, beta=1.0, log=True)
+        self.assertEqual(b1_log.get_neighbors(
+            0.99, rs=np.random.RandomState(42), transform=True), [751, 650, 454, 458])
+        self.assertEqual(b1_log.get_neighbors(
+            0.01, rs=np.random.RandomState(42), transform=True), [5, 6, 2, 9])
+
+        # Test attributes are accessible
+        self.assertEqual(f1.name, "param")
+        self.assertAlmostEqual(f1.alpha, 3.0)
+        self.assertAlmostEqual(f1.beta, 1.1)
+        self.assertAlmostEqual(f1.q, None)
+        self.assertEqual(f1.log, False)
+        self.assertEqual(f1.default_value, 2)
+        # beta parameters are normalized scaled to the unit cube, so for the parameter
+        # in [-2, 2]-range, the values are staggered from 0.1 to 0.9
+        #
+        self.assertAlmostEqual(f1.normalized_default_value, 0.9, places=4)
+
+        # Test copy
+        copy_f1 = copy.copy(f1)
+
+        self.assertEqual(copy_f1.name, f1.name)
+        self.assertEqual(copy_f1.alpha, f1.alpha)
+        self.assertEqual(copy_f1.beta, f1.beta)
+        self.assertEqual(copy_f1.default_value, f1.default_value)
+
+        f2 = BetaIntegerHyperparameter("param", lower=-2.0, upper=4.0, alpha=3.0, beta=1.1, q=2)
+        f2_ = BetaIntegerHyperparameter("param", lower=-2, upper=4, alpha=3, beta=1.1, q=2)
+        self.assertEqual(f2, f2_)
+
+        self.assertEqual(
+            "param, Type: BetaInteger, Alpha: 3.0 Beta: 1.1, Range: [-2, 4], Default: 4, "
+            "Q: 2", str(f2))
+
+        f3 = BetaIntegerHyperparameter("param", lower=1, upper=1000, alpha=3.0, beta=2.0, log=True)
+        f3_ = BetaIntegerHyperparameter("param", lower=1, upper=1000, alpha=3.0, beta=2.0, log=True)
+        self.assertEqual(f3, f3_)
+        self.assertEqual(
+            "param, Type: BetaInteger, Alpha: 3.0 Beta: 2.0, Range: [1, 1000], Default: 100, "
+            "on log-scale", str(f3))
+
+        with self.assertRaises(ValueError):
+            BetaIntegerHyperparameter("param", lower=-1, upper=10.0, alpha=6.0, beta=2.0, log=True)
+
+        with self.assertRaisesRegex(ValueError, "Illegal default value 0"):
+            BetaIntegerHyperparameter(
+                "param", lower=1, upper=10.0, alpha=3.0, beta=2.0, default_value=0, log=False)
+        with self.assertRaisesRegex(ValueError, "Illegal default value 0"):
+            BetaIntegerHyperparameter(
+                "param", lower=1, upper=1000.0, alpha=3.0, beta=2.0, default_value=0, log=True)
+        f4_legal_nolog = BetaIntegerHyperparameter(
+            "param", lower=1, upper=10.0, alpha=3.0, beta=2.0, default_value=1, log=True)
+        f4_legal_log = BetaIntegerHyperparameter(
+            "param", lower=1, upper=10.0, alpha=3.0, beta=2.0, default_value=1, log=False)
+
+        self.assertAlmostEqual(f4_legal_nolog.default_value, 1)
```

Comment:
Same as above, I'm surprised about this.

Reply:
Same response - I think it's up to us to decide what makes sense here. My reasoning is the following:

If you design a parameter on a log scale - with range from 1-100, with the default in the middle (at 10), you want to pass in 10, and not 1, as the default value - which would be the alternative. I simply think it is the more intuitive way to do it.